internal/processors/cloudeventconvert/cloudeventconvert.go

```diff
@@ -195,7 +196,25 @@ func (c *cloudeventProcessor) validateSignature(event *cloudevent.CloudEvent[jso
 		return false, fmt.Errorf("failed to unmarshal public key: %w", err)
 	}
 	recoveredAddress := crypto.PubkeyToAddress(*pubKey)
-	return common.HexToAddress(strings.TrimSpace(attestor)) == recoveredAddress, nil
+	if source != recoveredAddress {
+		return c.verifyERC1271Signature(signature, msgHash, source)
+	}
+
+	return true, nil
+}
+
+func (c *cloudeventProcessor) verifyERC1271Signature(signature []byte, msgHash []byte, source common.Address) (bool, error) {
+	contract, err := web3.NewErc1271(source, c.ethClient)
+	if err != nil {
+		return false, fmt.Errorf("failed to connect to address: %s: %w", source, err)
+	}
+
+	result, err := contract.IsValidSignature(nil, [32]byte(msgHash), signature)
```

Comment:
nitpick: As long as we're returning an error, we could check that len = 32.

This has no real chance of happening as written, since you pass in the result of keccak.

So sorry.